### DISCUSSION TITLE
Access to the vue-router from the outside

### DISCUSSION THREAD

**Discussion by darthsoup** [antfu/vite-ssg#190] [Q&A]
Hi all,

Is there a reason that only `RouterOptions` can be added to the ViteSSG setup function? In my projects I create the "router" before so that I can use it also outside of VueJS components (like axios interceptors). 

When I looked at the code, it seemed that this behaviour was intentional or maybe I don't understand SSG's approach correctly?!

Thanks

**Comment by jamus** [comment]
I had a simliar need. Essentially I’d like to access the router from Vuex and Xstate. Usually you’d import the router instance but I’m not sure how to do that (or if you should) when using Vite-ssg. Is there a way to 'Access to the vue-router from the outside'?

**Reply by khaledOghli** [reply]
do you find any solution how

**Reply by khaledOghli** [reply]
> I had a simliar need. Essentially I’d like to access the router from Vuex and Xstate. Usually you’d import the router instance but I’m not sure how to do that (or if you should) when using Vite-ssg. Is there a way to 'Access to the vue-router from the outside'?

maybe it's not the best way ,but i did that : 
let router;
export const createApp = ViteSSG(
  App,
  { routes, base: import.meta.env.BASE_URL },
  (ctx) => {
    router = ctx.router;
  },
)
export {router}


from api.js 
import { router } from '@/main.ts';

**Comment by borowskikrzysztof** [comment]
I have the same problem :( Somebody figured out this?

**Comment by khaledOghli** [comment]
any one figure out how to import router from outside.

**Comment by khaledOghli** [comment]
maybe it's not the best way ,but this what i did : 

```
let router;
export const createApp = ViteSSG(
  App,
  { routes, base: import.meta.env.BASE_URL },
  (ctx) => {
    router = ctx.router;
  },
)
export {router}

```

```
from api.js 
import { router } from '@/main.ts';

```

**Reply by tcyeee** [reply]
This change made the code look like a mess of yarn, but I couldn't find a better way ☹️

**Reply by Sunpacker** [reply]
@antfu +1
